src/Mesh/Interpolation.jl

```diff
@@ -8,207 +8,443 @@ using CLIMA.Mesh.Geometry
+    interpolate_local!(intrp_brck::InterpolationBrick{FT,T,FTV,FTVD,TVD,FTA2,UI8AD,UI16V,UI16VD,I32V},
+                              sv::AbstractArray{FT}, v::AbstractArray{FT}) where {FT <: AbstractFloat,
+                                                                                   T <: Int,
+                                                                                 FTV <: AbstractVector{FT}, 
+                                                                                FTVD <: AbstractVector{FT},
+                                                                                 TVD <: AbstractVector{T},
+                                                                                FTA2 <: Array{FT,2},
+                                                                               UI8AD <: AbstractArray{UInt8,2}, 
+                                                                               UI16V <: AbstractVector{UInt16},
+                                                                              UI16VD <: AbstractVector{UInt16},
+                                                                                I32V <:AbstractVector{Int32}}
 
 This interpolation function works for a brick, where stretching/compression happens only along the x, y & z axis.
 Here x1 = X1(ξ1); x2 = X2(ξ2); x3 = X3(ξ3)
 
 # input
  - `intrp_brck` Initialized InterpolationBrick structure
- - `sv` State vector
- - `st_idx` # of state vector variable to be interpolated
- - `poly_order` polynomial order for the simulation
+ - `sv` State Array consisting of various variables on the discontinuous Galerkin grid
+ - `v`  Interpolated variables
 """
-function interpolate_brick!(intrp_brck::InterpolationBrick, sv::AbstractArray{FT}, st_idx::T) where {T <: Integer, FT <: AbstractFloat}
-    qm1 = intrp_brck.poly_order + 1
-    Nel = length(intrp_brck.realelems)
-    m1_r, m1_w = GaussQuadrature.legendre(FT,qm1,GaussQuadrature.both)
-    wb = Elements.baryweights(m1_r)
-
-    vout    = FT(0)
-    vout_ii = FT(0)
-    vout_ij = FT(0)
-
-    phit = Vector{FT}(undef,qm1)
-
-    for el in 1:Nel #-----for each element elno 
-        if length(intrp_brck.ξ1[el]) > 0
-            l1 = length(intrp_brck.ξ1[el]); l2 = length(intrp_brck.ξ2[el]); l3 = length(intrp_brck.ξ3[el]) 
-            lag    = @view sv[:,st_idx,el]
-
-            phir = Elements.interpolationmatrix(m1_r, intrp_brck.ξ1[el], wb)
-            phis = Elements.interpolationmatrix(m1_r, intrp_brck.ξ2[el], wb)
-
-            for k in 1:l3 # interpolating point-by-point
-                interpolationvector_1pt!(m1_r,intrp_brck.ξ3[el][k],wb,phit)
-
-                for j in 1:l2, i in 1:l1
-                    vout = 0            
-                    for ik in 1:qm1
-                        v_ij = 0
-                        for ij in 1:qm1
-                            v_ii = 0
+function interpolate_local!(intrp_brck::InterpolationBrick{FT,T,FTV,FTVD,TVD,FTA2,UI8AD,UI16V,UI16VD,I32V},
```

Comment:
```suggestion
function interpolate_local!(intrp_brck::InterpolationBrick{FT},
```
and remove the parameters below (and in the docs)